src/long.ts

```diff
@@ -602,7 +603,7 @@ export class Long {
    * @param multiplier - Multiplier
    * @returns Product
    */
-  multiply(multiplier: string | number | Long): Long {
+  multiply(multiplier: string | number | Long | Timestamp): Long {
```

Comment:
This function will return a timestamp if passed one, is there a way we could pipe that information through without too much trouble? Should we just return a union of Long and Timestamp?

Reply:
Looks to be that all the mathematical methods construct and return a new Long object so the return types are correct, resolving.